docs/table/construct_table.rst

```diff
@@ -960,3 +964,71 @@ a `~astropy.table.Column` object with a ``unit`` attribute::
 
 To learn more about using standard `~astropy.table.Column` objects with defined
 units, see the :ref:`columns_with_units` section.
+
+
+Compatible tables
+^^^^^^^^^^^^^^^^^
+
+In order to improve interoperability between different table classes, an
+astropy |Table| object can be created directly from any other table
+object ``data`` that provides an ``__astropy_table__`` method.  In this case the
+``__astropy_table__`` method will be called as follows::
+
+  table = data.__astropy_table__(cls, copy)
+
+where ``cls`` is the |Table| class or subclass that is being instantiated,
+and ``copy`` indicates whether a copy of the values in ``data`` should be
+provided.  If ``copy`` is ``False`` then the ``__astropy_table__`` method
```

Comment:
This discussion is exactly the reason I added the `unviewable` argument on lsst/afw#64; it controls what happens when a copy is required for a column, but `copy=False` (the choices are raise an exception, skip the column, or copy the column).  It sounds like it *might* be appropriate for that argument (or an equivalent) to actually be part of the `__init__` here, with the current behavior implying a default of "raise" and the proposal here being to change it to "copy".